astropy/coordinates/tests/test_frames.py

```diff
@@ -1473,3 +1487,25 @@ def test_galactocentric_references(reset_galactocentric_defaults):
                 assert k not in galcen_custom.frame_attribute_references
             else:
                 assert k in galcen_custom.frame_attribute_references
+
+
+def test_galactocentric_registry():
+
+    from astropy.coordinates import (ICRS,
+                                     galactocentric_frame_defaults)
+
+    galactocentric_frame_defaults.register(
+        name="test",
+        value={
+            "galcen_coord": ICRS(ra=266.0 * u.degree, dec=-29 * u.degree),
+            "galcen_distance": 8.1 * u.kpc,
+            "galcen_v_sun": r.CartesianDifferential([12.9, 245.6, 7.8] * u.km / u.s),
+            "z_sun": 21.0 * u.pc,
+            "roll": 0 * u.deg},
+        references=None
+    )
```

Comment:
The test should probably include that the frame is in fact registered. Also, do put everything in a `try/finally` so that in the end one can be sure the state is back to what it was originally (even if the test failed).

Reply:
done.

Reply:
Ideally, the `try` should be before the registering is done (guard against bug in registration function; yes, it is a bit over the top!)